yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -38,6 +38,7 @@ import org.apache.spark.{SparkEnv, Logging, SecurityManager, SparkConf}
 import org.apache.spark.deploy.yarn.YarnSparkHadoopUtil._
 import org.apache.spark.scheduler.cluster.CoarseGrainedSchedulerBackend
 import org.apache.spark.util.AkkaUtils
+import java.lang.reflect.Constructor
```

Comment:
Nit, import ordering here.

This looks like a pretty straightforward change, yeah. Too bad this requires reflection, but it's clean. I look forward to being able to assume later versions of YARN / Hadoop to undo some of this. (Although requiring 2.6 probably won't happen too soon.)

The key use case this enables is, for example, running a Spark app on a subset of a cluster's machines that are labeled as having a GPU. I like this given the relatively low impact simple change and the upside to what it enables.